module/zfs/zvol.c

```diff
@@ -38,6 +38,33 @@
  * Copyright (c) 2016 Actifio, Inc. All rights reserved.
  */
 
+/*
+ * Note on locking of zvol state structures.
+ *
+ * These structures are used to maintain internal state used to emulate block
+ * devices on top of zvols. In particular, management of device minor number
+ * operations - create, remove, rename, and set_snapdev - involves access to
+ * these structures. The zvol_state_lock is primarily used to protect the
+ * zvol_state_list,  while the zv_state_lock is used to protect the contents
```

Comment:
The names `zvol_state_lock` and `zv_state_lock` are so close how about adding `zv->` here to make it clear it's a member of the structure.  And I think we can break this in to two sentences.

```diff
- * zvol_state_list,  while the zv_state_lock is used to protect the contents
+ * zvol_state_list.  The zv->zv_state_lock is used to protect the contents
```